CDAP-13558 Support CDC pipelines on Spark version 2.x

mlozbin-cybervisiontech opened the pull request:
JIRA - https://issues.cask.co/browse/CDAP-13558
WIKI - https://wiki.cask.co/display/CE/CDC+Solution+Enhancements

In scope of this PR:
- implemented integration tests for CTSQLServer source plugin
- introduced OperationType enum to make CDC records more type safe

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- implemented integration tests for CTSQLServer source plugin
- introduced OperationType enum to make CDC records more type safe

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- implemented integration tests for CTSQLServer source plugin
- introduced OperationType enum to make CDC records more type safe

mlozbin-cybervisiontech commented:
@albertshau, could you please take a look.

albertshau commented on docker-compose/cdc-env/docker-compose.yml:
```diff
@@ -0,0 +1,9 @@
+version: '3'
```
is this intentionally included?

albertshau replied:
I see this is what the unit test expects to be used.

albertshau replied:
Can you add some lines to the README about how to run the tests?

albertshau commented on src/test/java/co/cask/cdc/plugins/integration/source/CTSQLServerPluginTest.java:
```diff
@@ -0,0 +1,410 @@
+  }
+
+  private static Connection getConnectionToRoot() throws SQLException {
+    String connectionString = String.format("jdbc:sqlserver://%s:%s", HOST, PORT);
```
is this assuming there is a sqlserver running on local host? It would be better to have the host, port, user, and password all configurable through system properties.

albertshau commented on src/test/java/co/cask/cdc/plugins/integration/source/CTSQLServerPluginTest.java:
```diff
@@ -0,0 +1,410 @@
+      ETLPlugin sinkConfig = MockSink.getPlugin(outputTable);
+
+      programManager = deployETL(sourceConfig, sinkConfig, "testChangesSource");
+      programManager.start();
```
though there is nothing wrong here, use startAndWaitForRun() because it is generally less easy to mis-use than waitForRun(). We might remove waitForRun() at some point in the future.

mlozbin-cybervisiontech committed:
CDAP-13558 Support CDC pipelines on Spark version 2.x
- implemented integration tests for CTSQLServer source plugin
- introduced OperationType enum to make CDC records more type safe

mlozbin-cybervisiontech commented:
@albertshau fixed all comments

albertshau reviewed (approved):
lgtm
